Allow bookinfo services domain to be configurable

zlav opened the pull request:
This is needed to run on CloudFoundry

Signed-off-by: Zachary Gershman <zgershman@pivotal.io>

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: 
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **frankbu**

Assign the PR to them by writing `/assign @frankbu` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[samples/OWNERS](https://github.com/istio/istio/blob/master/samples/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["frankbu"]} -->

istio-testing commented:
Hi @zlav. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

zlav commented:
/assign @frankbu

frankbu commented on samples/bookinfo/src/productpage/productpage.py:
```diff
@@ -44,26 +45,29 @@
 Bootstrap(app)
 
+servicesDomain = "" if (os.environ.get("SERVICES_DOMAIN") == None) else "." + os.environ.get("SERVICES_DOMAIN")
+servicesPort = "9080"
```
why is servicesPort moved to a variable if it's not configurable anyway?

frankbu commented on samples/bookinfo/src/productpage/productpage.py:
```diff
@@ -44,26 +45,29 @@
+
 details = {
-    "name" : "http://details:9080",
+    "name" : "http://{0}{1}:{2}".format("details", servicesDomain, servicesPort),
```
nit, more clear like this: "http://details{0}:{1}".format(servicesDomain, servicesPort)

zlav commented:
Hi @frankbu The googlebot removed the cla:yes tag after I added another committer but I removed them so it should be all good. I was initially verified in the beginning of the request and if you need to reverify that I signed it I can send you the verification

aaronshurley commented:
@frankbu please let us know if there's anything else to do. I don't think our changes would've affected codecov or racetest

rshriram commented:
/ok-to-test

rshriram commented:
as part of this change, can you also change all the bookinfo.yamls to point to the new version of all bookinfo images? changes to bookinfo code are not tested, until they are docker-fied

istio-testing commented:
@zlav: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests-v1alpha3.sh | 8fe7bdf2b72794d845117073c79572f5b31feb31 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/101/) | `/test e2e-bookInfo-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
New changes are detected. LGTM label has been removed.

zachgersh commented:
/ok-to-test

istio-testing commented:
@zlav: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests-v1alpha3.sh | 0337a688a9c5ea45c36dae94194e2a69ad1572ef | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/747/) | `/test e2e-bookInfo-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@zlav: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests-v1alpha3.sh | 0337a688a9c5ea45c36dae94194e2a69ad1572ef | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/747/) | `/test e2e-bookInfo-envoyv2-v1alpha3`
prow/e2e-bookInfoTests.sh | 0337a688a9c5ea45c36dae94194e2a69ad1572ef | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests/3772/) | `/test e2e-bookInfo`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@zlav: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests-v1alpha3.sh | 0337a688a9c5ea45c36dae94194e2a69ad1572ef | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/747/) | `/test e2e-bookInfo-envoyv2-v1alpha3`
prow/e2e-bookInfoTests.sh | 0337a688a9c5ea45c36dae94194e2a69ad1572ef | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests/3772/) | `/test e2e-bookInfo`
prow/istio-pilot-e2e.sh | 0337a688a9c5ea45c36dae94194e2a69ad1572ef | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/istio-pilot-e2e/6911/) | `/test istio-pilot-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@zlav: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests.sh | 0337a688a9c5ea45c36dae94194e2a69ad1572ef | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests/3772/) | `/test e2e-bookInfo`
prow/istio-pilot-e2e.sh | 0337a688a9c5ea45c36dae94194e2a69ad1572ef | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/istio-pilot-e2e/6911/) | `/test istio-pilot-e2e`
prow/e2e-bookInfoTests-v1alpha3.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/749/) | `/test e2e-bookInfo-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@zlav: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e.sh | 0337a688a9c5ea45c36dae94194e2a69ad1572ef | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/istio-pilot-e2e/6911/) | `/test istio-pilot-e2e`
prow/e2e-bookInfoTests-v1alpha3.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/749/) | `/test e2e-bookInfo-envoyv2-v1alpha3`
prow/e2e-bookInfoTests.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests/3774/) | `/test e2e-bookInfo`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@zlav: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests-v1alpha3.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/749/) | `/test e2e-bookInfo-envoyv2-v1alpha3`
prow/e2e-bookInfoTests.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests/3774/) | `/test e2e-bookInfo`
prow/istio-pilot-e2e.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/istio-pilot-e2e/6913/) | `/test istio-pilot-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

zachgersh commented:
/test e2e-bookInfo-envoyv2-v1alpha3

istio-testing commented:
@zlav: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests/3774/) | `/test e2e-bookInfo`
prow/istio-pilot-e2e.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/istio-pilot-e2e/6913/) | `/test istio-pilot-e2e`
prow/e2e-bookInfoTests-v1alpha3.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/751/) | `/test e2e-bookInfo-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@zlav: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests/3774/) | `/test e2e-bookInfo`
prow/e2e-bookInfoTests-v1alpha3.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/751/) | `/test e2e-bookInfo-envoyv2-v1alpha3`
prow/istio-pilot-e2e.sh | 0437e056ab7f543ca8a7fa762b1c2501698782b5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/istio-pilot-e2e/6917/) | `/test istio-pilot-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@zlav: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-bookInfoTests.sh | e8b599b0de232f9d93f2f90c1fad217540bb25b8 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests/3774/) | `/test e2e-bookInfo`
prow/istio-pilot-e2e.sh | 0437e056ab7f543ca8a7fa762b1c2501698782b5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/istio-pilot-e2e/6917/) | `/test istio-pilot-e2e`
prow/e2e-bookInfoTests-v1alpha3.sh | 0437e056ab7f543ca8a7fa762b1c2501698782b5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/754/) | `/test e2e-bookInfo-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@zlav: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e.sh | 0437e056ab7f543ca8a7fa762b1c2501698782b5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/istio-pilot-e2e/6917/) | `/test istio-pilot-e2e`
prow/e2e-bookInfoTests-v1alpha3.sh | 0437e056ab7f543ca8a7fa762b1c2501698782b5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests-envoyv2-v1alpha3/754/) | `/test e2e-bookInfo-envoyv2-v1alpha3`
prow/e2e-bookInfoTests.sh | 0437e056ab7f543ca8a7fa762b1c2501698782b5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4798/e2e-bookInfoTests/3778/) | `/test e2e-bookInfo`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

zachgersh commented:
Closed in favor of #5483
